src/wash/hooks/useMint.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useState } from 'react';
+import { useUmi } from '../contexts/useUmi';
+import { WASH_ENDPOINT_ROOT_URI } from '../utils/constants';
+import { ChainType } from '@lifi/sdk';
+import { useAccount } from '@lifi/wallet-management';
+import { base58 } from '@metaplex-foundation/umi/serializers';
+
+export type TUseMint = {
+  onMint: VoidFunction;
+  mintStatus: string;
+  isMinting: boolean;
+  error?: string;
+};
+
+export function useMint(refetchNft?: VoidFunction): TUseMint {
+  const [isMinting, set_isMinting] = useState(false);
+  const [error, set_error] = useState<string | undefined>();
+  const [mintStatus, set_mintStatus] = useState('');
+  const { umi } = useUmi();
+  const { account } = useAccount({ chainType: ChainType.SVM });
+
+  const onMint = useCallback(async (): Promise<void> => {
+    if (!umi || !account.isConnected || !account.address) {
+      console.error('Wallet not connected or Umi not initialized');
+      return;
+    }
+
+    /******************************************************************************************
+     * Prepare the NFT mint, we need this because the backend will sign the authorization to
+     * reveal the NFT and the user will use that signature to mint the NFT
+     ******************************************************************************************/
+    let responsePrepareNft: Response | undefined = undefined;
+    try {
+      set_isMinting(true);
+      responsePrepareNft = await fetch(
+        `${WASH_ENDPOINT_ROOT_URI}/mint/prepare`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            userPublicKey: umi.identity.publicKey,
+          }),
+        },
+      );
+    } catch (err) {
+      set_isMinting(false);
+      set_error(
+        err instanceof Error ? err.message : 'An error occured while minting',
+      );
+      console.error('Error minting NFT:', error);
+    }
+
+    /******************************************************************************************
+     * Send the transaction to the Solana RPC
+     ******************************************************************************************/
+    let signature: Uint8Array | undefined = undefined;
+    let data: { tx: ArrayBufferLike; assetAddress: string } | undefined =
+      undefined;
+    try {
+      if (!responsePrepareNft) {
```

Comment:
Because of how fetch is working, response will still be there but data might not be